great_expectations/core/expectation_suite.py

```diff
@@ -249,9 +250,20 @@ def save(self) -> None:
         self._store.update(key=key, value=self)
 
     def is_added(self) -> ExpectationSuiteAddedDiagnostics:
-        return ExpectationSuiteAddedDiagnostics(
+        diagnostics = ExpectationSuiteAddedDiagnostics(
             errors=[] if self.id else [ExpectationSuiteNotAddedError(name=self.name)]
         )
+        if not diagnostics.is_added:
+            return diagnostics
+        return self._is_fresh()
+
+    def _is_fresh(self) -> ExpectationSuiteAddedDiagnostics:
+        key = self._store.get_key(name=self.name, id=self.id)
+        suite_dict = self._store.get(key=key)
+        suite = ExpectationSuite(**suite_dict)
```

Comment:
There is a comment about this in `suite_factory.py.get()`. There are errors that can happen here because, even though the suite, has been validated with marshmallow, the expectations inside have not been validated. Doing this will instantiate the expectations and do the pydantic validation.
So, if we need to do this here, we need to be a bit careful and handle `PydanticValidationError`s that can occur. See https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/core/factory/suite_factory.py#L115